include/openssl/md2.h

```diff
@@ -19,32 +19,37 @@
 # include <openssl/opensslconf.h>
 
 # ifndef OPENSSL_NO_MD2
-# include <stddef.h>
-# ifdef  __cplusplus
+#  include <stddef.h>
+#  ifdef  __cplusplus
 extern "C" {
-# endif
+#  endif
+#  if !defined(OPENSSL_NO_DEPRECATED_3_0)
 
 typedef unsigned char MD2_INT;
 
-# define MD2_DIGEST_LENGTH       16
-# define MD2_BLOCK               16
+#   define MD2_DIGEST_LENGTH       16
+#   define MD2_BLOCK               16
 
 typedef struct MD2state_st {
     unsigned int num;
     unsigned char data[MD2_BLOCK];
     MD2_INT cksm[MD2_BLOCK];
     MD2_INT state[MD2_BLOCK];
 } MD2_CTX;
-
-const char *MD2_options(void);
-int MD2_Init(MD2_CTX *c);
-int MD2_Update(MD2_CTX *c, const unsigned char *data, size_t len);
-int MD2_Final(unsigned char *md, MD2_CTX *c);
-unsigned char *MD2(const unsigned char *d, size_t n, unsigned char *md);
-
-# ifdef  __cplusplus
+#  endif
+
+DEPRECATEDIN_3_0(const char *MD2_options(void))
+DEPRECATEDIN_3_0(int MD2_Init(MD2_CTX *c))
+DEPRECATEDIN_3_0(int MD2_Update(MD2_CTX *c, const unsigned char *data,
+                                size_t len))
+DEPRECATEDIN_3_0(int MD2_Final(unsigned char *md, MD2_CTX *c))
+DEPRECATEDIN_3_0(unsigned char *MD2(const unsigned char *d, size_t n,
+                 unsigned char *md))
+
+#  if !defined(OPENSSL_NO_DEPRECATED_3_0)
```

Comment:
This looks weird, the initial `extern "C" `is not inside `!defined(OPENSSL_NO_DEPRECATED_3_0)`

Reply:
Missed this, the `extern "C"` was originally inside the `!defined(OPENSSL_NO_DEPRECATED_3_0)` but I moved it later after @levitte asked.

Reply:
Fixed now.